### PR TITLE
feat: [PL-41754]: Introducing Mongo FCV upgrade job

### PR DESCRIPTION
**Pull request by arya-harness**
SMP Upgrade preparation for mongo 6.
It requires a step that upgrades the mongo server atleast to a compatible version before proceeding.
Included the same step as part of pre-upgrade hook, so it can complete before proceeding.
Tests:
Pasted Graphic

upgrade will fail with the following error if cg-manager is not in right version:
image

### CODE REVIEW

File: src/harness/README.md [hunk 1]
```diff
@@ -290,8 +295,9 @@ plugins/s3:1.3.0
 | global.monitoring | object | `{"enabled":false,"path":"/metrics","port":8889}` | Enable monitoring for all harness services: disabled by default |
 | global.ng | object | `{"enabled":true}` | Enable to install NG (Next Generation Harness Platform) |
 | global.ngcustomdashboard | object | `{"enabled":false}` | Enable to install Next Generation Custom Dashboards (Beta) |
-| global.opa | object | `{"enabled":false}` | Enable to install Open Policy Agent (OPA) |
+| global.opa | object | `{"enabled":true}` | Default Enabled, As required by multiple services now (OPA) |
```
Comment (bharatgoelharness) [inline src/harness/README.md:298]: OPA is now derived from CD flag. So we should remove this. @avaakash

File: src/harness/README.md [hunk 2]
```diff
@@ -290,8 +295,9 @@ plugins/s3:1.3.0
...
 | global.overrideValidation | object | `{"restructuredValues":false}` | Enable to disable validation checks |
+| global.pdb.create | bool | `false` |  |
```
Comment (bharatgoelharness) [inline src/harness/README.md:300]: pdb will be true by default

Comment (bharatgoelharness) [reply src/harness/README.md:300]: please add the description as well

**Comment by bharatgoelharness** [comment]
Do we have customers who are on Mongo 4? 
And this script will be part of 0.22.0 then customers upgrading to 0.22.0 SMP can use Mongo 5 who are on Mongo 4?
Do we have to bake mongo 5 and mongo 6 images to 0.22.0?

File: src/harness/README.md (hunk 2 above)
Comment (arya-harness) [reply src/harness/README.md:300]: this is auto-generated Readme though

File: src/harness/templates/mongo-preupgradejob.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+              name: harness-secrets
+              key: mongodbUsername
+        - name: MONGO_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mongodb-replicaset-chart
+              key: mongodb-root-password
+        - name: MONGO_URI
+          value: 'mongodb://$(MONGO_USER):$(MONGO_PASSWORD)@mongodb-replicaset-chart-0.mongodb-replicaset-chart/?authSource=admin'
+      restartPolicy: Never
```
Comment (arya-harness) [inline src/harness/templates/mongo-preupgradejob.yaml:60]: make sure that securityContext is same as mongodb.

Comment (arya-harness) [reply src/harness/templates/mongo-preupgradejob.yaml:60]: Add service account.

Comment (arya-harness) [reply src/harness/templates/mongo-preupgradejob.yaml:60]: add RN to upgrade their external mongo to 6 as well.

**Comment by arya-harness** [comment]
Verification tests to do:
0.16.0 to 0.22.0
0.17.0 to 0.22.0

File: src/harness/templates/mongo-preupgradejob.yaml (hunk 3 above)
Comment (arya-harness) [reply src/harness/templates/mongo-preupgradejob.yaml:60]: argo CD different instructions

File: src/harness/values.yaml [hunk 4]
```diff
@@ -565,3 +565,11 @@ sto:
...
     nodeSelector: {}
     tolerations: []
+
+upgrades:
+  mongoFCVUpgrade:
+    enabled: true
+    image:
+      registry: docker.io
+      repository: bitnami/mongodb
+      tag: 5.0.24
```
Comment (arya-harness) [inline src/harness/values.yaml:575]: update to 6.0.13

**Comment by arya-harness** [comment]
Validated test 1: version 0.14.6 to upgrade 0.22.0
image

**Comment by arya-harness** [comment]
Verified case 2:
0.17.0 to 0.22.0 upgrade
Pasted Graphic 22
Job succeeds with 
HARNESS
More details to be added in Release notes on edge cases.

File: src/harness/templates/mongo-preupgradejob.yaml (hunk 3 above)
Comment (arya-harness) [reply src/harness/templates/mongo-preupgradejob.yaml:67]: Done.

**Review by avaakash — APPROVED** [review]

**Pull request merged**